pulpcore/cli/openapi.py

```diff
@@ -32,7 +32,12 @@ def __init__(
             "Accept": "application/json",
         }
         self._session = requests.session()
-        self._session.auth = (username, password)
+        if username and password:
+            self._session.auth = (username, password)
+        elif username and not password:
+            raise Exception("Password is required if username is set.")
+        elif password and not username:
```

Comment:
```suggestion
        elif password:
```
Technically we don't need it...